lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -2103,70 +2105,275 @@ static std::string to_string(VersionNumber Version) {
   return to_string(static_cast<uint32_t>(Version));
 }
 
-std::istream &operator>>(std::istream &I, SPIRVModule &M) {
-  SPIRVDecoder Decoder(I, M);
-  SPIRVModuleImpl &MI = *static_cast<SPIRVModuleImpl *>(&M);
-  // Disable automatic capability filling.
+namespace {
```

Comment:
I think it might be a good idea to start populating a new file for this functionality SPIRVParser.cpp. You can do that in this PR or the next.

Thanks

Reply:
Basing on my current investigation, when we don't have separated building blocks of parser and `SPIRVModule` (`getEntry` problem) this kind of split will yield more trouble than it's worth. Please note that `SPIRVModule` fields like SPIRVVersion are private and `operator>>` is a kind of factory/constructor. If I would try to separate it now then code changes would be more significant which would yield PR larger.

Reply:
Agreed. We can attempt this code refactoring at a later stage when things are more amenable to this change. Thanks